app/main.py

```diff
@@ -1,25 +1,89 @@
-from abc import ABC
+from __future__ import annotations
+from abc import ABC, abstractmethod
+from typing import Any, Type
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Type[Any], name: str) -> None:
+        self.protected = "_" + name
+
+    def __get__(self, instance: Any, owner: Type[Any]) -> Any:
+        return getattr(instance, self.protected)
+
+    def __set__(self, instance: Any, value: int) -> None:
+        setattr(instance, self.protected, value)
 
 
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(self, age: int, weight: int, height: int) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
+        self.validate()
+
+    @abstractmethod
+    def validate(self) -> None:
+        pass
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    def validate(self) -> None:
+        if not (4 <= self.age <= 14):
+            raise ValueError(
+                "Age should be between 4 and 14 for children."
+            )
+        if not (80 <= self.height <= 120):
+            raise ValueError(
+                "Height should be between 80 and 120 for children."
+            )
+        if not (20 <= self.weight <= 50):
+            raise ValueError(
+                "Weight should be between 20 and 50 for children."
+            )
```

Comment:
Your validation should be in your `__set__ `method ) You just will need to call instances of your `IntegerRange`  class